elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/SQLQueryEngine.java

```diff
@@ -279,4 +294,24 @@ public static String generateColumnReference(Path path, EntityDictionary diction
     public static String getClassAlias(Class<?> entityClass) {
         return FilterPredicate.getTypeAlias(entityClass);
     }
+
+    /**
+     * Generate alias for represent the join path.
+     * The path would start with the class alias of the first element, and then each field would append "_fieldName" to
+     * the result.
+     * The last element would not be included as that's not a part of the join path.
+     *
+     * @param path path that represents a relationship chain
+     * @return join path alias, i.e. <code>foo.bar.baz</code> would be <code>foo_bar</code>
+     */
+    public static String getJoinPathAlias(Path path) {
```

Comment:
Can this be unified with FilterPredicate logic?

Reply:
👍